client/src/components/ManageRequestsTable.js

```diff
@@ -1,20 +1,184 @@
+import unique from 'helpers/unique'
+import { resourceCategories } from 'components/resources'
+import api from 'api'
 
-export default ({ requests }) => {
+const RequestFilters = ({
+  requests,
+  filters = ['status', 'assignedTo', 'resourceType', 'team'],
+  onFilterChange
+}) => {
+  const [fetchedTeams, setFetchedTeams] = React.useState(false)
+  const [teamOptions, setTeamOptions] = React.useState([])
+  const [memberOptions, setMemberOptions] = React.useState([])
+  const resourceTypeOptions = [
+    { label: 'All', value: undefined },
+    ...resourceCategories.map((value) => ({
+      label: getReadable(value),
+      value
+    }))
+  ]
+  const statusOptions = [
+    { label: 'All', value: undefined },
+    ...allStatuses.map((value) => ({
+      label: getReadable(value),
+      value
+    }))
+  ]
+
+  const { user, token } = useUser()
+  const [status, setStatus] = React.useState()
+  const [assignedTo, setAssignedTo] = React.useState()
+  const [resourceType, setResourceType] = React.useState()
+  const [team, setTeam] = React.useState()
+
+  React.useEffect(() => {
+    const fetchTeams = async () => {
+      setFetchedTeams(true)
+      const teamsRequestsPromises = user.organizations.map((userTeam) =>
+        api.teams.get(userTeam.id, token)
+      )
+      const teamsRequests = await Promise.all(teamsRequestsPromises)
+
+      const newTeams = teamsRequests.map((teamRequest) => {
+        if (teamRequest.isOk)
+          return {
+            label: teamRequest.response.name,
+            value: teamRequest.response.id
+          }
+        return null
+      })
+
+      const cleanedNewTeams = newTeams.filter((t) => t !== null)
+      setTeamOptions([{ label: 'All', value: undefined }, ...cleanedNewTeams])
+
+      const newMembers = [].concat(
+        ...teamsRequests.map((teamRequest) => {
+          if (teamRequest.isOk)
+            return teamRequest.response.members.map((m) => ({
+              label: `${m.full_name} | ${m.email}`,
+              value: m.id
+            }))
+          return []
+        })
+      )
+
+      const cleanedNewMembers = unique(newMembers, 'value')
+
+      setMemberOptions([
+        { label: 'All', value: undefined },
+        ...cleanedNewMembers
+      ])
+    }
+    if (!fetchedTeams) fetchTeams()
+  })
+
+  React.useEffect(() => {
+    const filteredStatus = requests.filter(
+      (r) => status === undefined || getRequestStatus(r) === status
+    )
+
+    const filteredResourceType = filteredStatus.filter(
+      (r) => !resourceType || r.material.category === resourceType
+    )
+
+    const filteredAssignedTo = filteredResourceType.filter(
+      (r) => !assignedTo || r.assigned_to === assignedTo
+    )
+
+    const filteredTeam = filteredAssignedTo.filter(
+      (r) => !team || r.material.organization.id === team
+    )
+
+    const filteredRequests = [...filteredTeam]
+
+    const cleanedFilteredRequests = unique(filteredRequests, 'id')
```

Comment:
Should the sorting to make sure that they're unique happen before you filter them on all the other attributes?

Reply:
It actually won't really affect the filtered requests here since they are essentially chained. This was more for if we wanted to do OR logic instead of AND which would be if we filtered each one from the original list and then merged that list. But also depending on the source of the requests it is an additional check that has not side effect other than the list will not have dupes